x-pack/plugins/apm/server/lib/search_strategies/queries/query_failure_correlation.ts

```diff
@@ -101,8 +108,25 @@ export const fetchFailedTransactionsCorrelationPValues = async (
       successPercentage:
         (bucket.bg_count - bucket.doc_count) /
         (overallResult.bg_count - overallResult.doc_count),
+      histogram: [] as Array<{ key: number; doc_count: number }>,
     };
   });
 
+  for (let i = 0; i < result.length; i++) {
+    const current = result[i];
+
+    const histogram = await fetchTransactionDurationRanges(
+      esClient,
+      params,
+      histogramRangeSteps,
+      [
+        { fieldName: EVENT_OUTCOME, fieldValue: EventOutcome.failure },
+        { fieldName: current.fieldName, fieldValue: current.fieldValue },
+      ]
+    );
+
+    current.histogram = histogram;
+  }
```

Comment:
Are you using a `for` loop to ensure the promises are resolved in sequence? And what does `current.histogram` do?

Reply:
This was a bit of a leftover from experimentation. I refactored it to get rid of the `current` and use `for of` in 
95f4a49. It's sequential by intention because we don't know how many calls there could be to avoid hammering ES.